components/playlist/playlist_download_request_manager.cc

```diff
@@ -145,11 +145,19 @@ void PlaylistDownloadRequestManager::GetMedia(content::WebContents* contents) {
       media_detector_component_manager_->GetMediaDetectorScript();
   DCHECK(!media_detector_script.empty());
 
+#if BUILDFLAG(IS_ANDROID)
+  content::RenderFrameHost::AllowInjectingJavaScript();
+  contents->GetPrimaryMainFrame()->ExecuteJavaScript(
+      base::UTF8ToUTF16(media_detector_script),
+      base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
+                     weak_factory_.GetWeakPtr(), contents->GetWeakPtr()));
```

Comment:
The media object that is passed to this callback can be arbitrary content (since the page can override javascript prototypes as needed in order to return whatever it wants). Can you add some strict checking that the media is a Dict of this form? https://github.com/brave/brave-core/pull/16419/files#diff-0ed16672d3d0c3e4028da7cc52a14a6d436d4f9e3a3c1dae9ee599887e3b1ce7R183-R194

Reply:
also can we make sure that string fields like name/title/etc only contain alphanumeric+whitespace characters?

Reply:
and URL fields must have scheme http or https

Reply:
> also can we make sure that string fields like name/title/etc only contain alphanumeric+whitespace characters?

Maybe expand this to ASCII only? Many youtube videos have non alphanumeric characters. Duration field should be `int` only though.
@sangwoo108 also, since this script is only needed for YT, can we add a check to inject only for *.youtube.com domains?

Reply:
> Can you add some strict checking that the media is a Dict of this form?

I think we're checking if it's dict. And default type checking is also done here.

https://github.com/brave/brave-core/blob/0eb384c3a5f012f5f236ce32490beb264dfcecc2/components/playlist/playlist_download_request_manager.cc#L193


> name/title/etc only contain alphanumeric+whitespace (or ASCII)

I think we can't apply this checks. Contents could be CJK or other languages.

> URL fields must have scheme http or https

Yep, good call.

Reply:
thanks for that. couple of comments:

1. https://github.com/brave/brave-core/pull/16419/files#diff-0ed16672d3d0c3e4028da7cc52a14a6d436d4f9e3a3c1dae9ee599887e3b1ce7R195 - it seems detected field is not actually used anywhere?
2. where is the check that the domain is `*.youtube.com` before injecting?
3. for security checks like https://github.com/brave/brave-core/blob/0eb384c3a5f012f5f236ce32490beb264dfcecc2/components/playlist/playlist_download_request_manager.cc#L53, should it be CHECK instead of DCHECK? in other words we would rather have the browser crash than do something that is very insecure.

Reply:
> it seems detected field is not actually used anywhere?

Yeah, in this PR the use cases are not included. The fields like author, duration were requested by Android.

> where is the check that the domain is *.youtube.com before injecting?

Let me leave pointers.
[1]
[2]

> in other words we would rather have the browser crash than do something that is very insecure.

That makes sense. Changed.